forum/users/validators.py

```diff
@@ -0,0 +1,46 @@
+import re
+from string import punctuation
+from rest_framework.exceptions import ValidationError
+from rest_framework import serializers
+
+class CustomUserValidator:
+    @staticmethod
+    def validate_passwords_match(password, password2):
+        if password != password2:
+            raise serializers.ValidationError({"Error": "Passwords don't match"})
+        
+    @staticmethod
+    def validate_password(password):
+        try:
+            min_length = 8
+            char_types = {
+                'digit': re.compile(r'\d'),
+                'upper': re.compile(r'[A-Z]'),
+                'lower': re.compile(r'[a-z]'),
+                'symbol': re.compile(r'[{}]'.format(re.escape(punctuation)))
+            }
+
+            errors = []
+            if len(password) < min_length:
+                errors.append('Password must be at least 8 characters long.')
+
+            for char_type, pattern in char_types.items():
+                if not pattern.search(password):
+                    errors.append(f'Password must contain at least one {char_type}.')
+
+            if errors:
+                raise ValidationError(' '.join(errors))
+            return password 
+        except ValidationError as e:
+            raise serializers.ValidationError({"Error": e.detail})
```

Comment:
The `try-except` blocks are not needed if you directly raise `serializers.ValidationError`.

Reply:
Done, thanks